app/frontend/src/features/userPage/UserPlace.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import { SmokingLocation, User } from "../../pb/api_pb";
+import { makeStyles, Typography } from "@material-ui/core";
+import TextBody from "../../components/TextBody";
+import { smokingLocationLabels } from "../../constants";
+
+const useStyles = makeStyles((theme) => ({
+  list: {
+    listStyle: "none",
+    padding: 0,
+    marginBlockStart: 0,
+    //workaround for https://github.com/cssinjs/jss/issues/1414
+    //still present despite upgrading to 10.5.0
+    marginBlockEnd: `${theme.spacing(1)}px`,
+  },
+  listItem: {},
+  hostingPreferenceResponse: {
+    fontWeight: "bold",
+  },
+  subheading: {
+    fontSize: "1.25rem",
+  },
+}));
+
+export default function UserPlace({ user }: { user: User.AsObject }) {
+  const classes = useStyles();
+
+  const preferences = [
```

Comment:
Not sure how I feel about this - I don't think it's necessarily clearer than just writing out the JSX out in repeat instead of doing this to be used for generating the needed JSX with map. I think mapping over array of data to JSX makes more sense for dynamic data.

Reply:
And, you could also prepare same variables in the `.map` iteration before you `return` the `jsx`.

Reply:
I made a component instead